src/ape_ethereum/ecosystem.py

```diff
@@ -320,6 +320,23 @@ def encode_calldata(self, abi: Union[ConstructorABI, MethodABI], *args) -> HexBy
         if not abi.inputs:
             return HexBytes("")
 
+        # Convert dicts and other objects to struct inputs
+        args = list(args)  # type: ignore[assignment]
+        for idx, ipt in enumerate(abi.inputs):
+            arg = args[idx]
+            if (
+                ipt.type == "tuple"
+                and ipt.components
+                and all(m.name for m in ipt.components)
+                and not isinstance(arg, tuple)
+            ):
+                if isinstance(arg, dict):
+                    args[idx] = tuple([arg[m.name] for m in ipt.components])  # type: ignore[index]
+
+                else:
+                    arg = [getattr(arg, m.name) for m in ipt.components]  # type: ignore[arg-type]
+                    args[idx] = tuple(arg)  # type: ignore[index]
+
         input_types = [i.canonical_type for i in abi.inputs]
```

Comment:
do we also need to handle nested / recursive structures?

Reply:
I think `.canonical_type` is recursive

Reply:
https://github.com/ApeWorX/ethpm-types/blob/main/ethpm_types/abi.py#L58

Reply:
I moreso meant struture values as inputs, like nested dicts with account objects in the middle:

```
account = accounts[0]
argument = {"foo": {"bar": account}}
contract.my_method(arguments, sender=account)
```